services/weblate/base-weblate-service.js

```diff
@@ -0,0 +1,21 @@
+import Joi from 'joi'
+import { BaseJsonService } from '../index.js'
+import { optionalUrl } from '../validators.js'
+
+export default class BaseWeblateService extends BaseJsonService {
+  static queryParamSchema = Joi.object({
+    server: optionalUrl,
+  }).required()
+
+  static auth = {
+    passKey: 'weblate_api_key',
+    authorizedOrigins: ['https://hosted.weblate.org'],
```

Comment:
Will need to remove this. We want Shields server hosters (be it self-hosters or us for Shields.io) to specify which Weblate instances are authorized via the standard configuration options. For the Shields.io runtime we will set this value to `hosted.weblate.org` along with setting our token, but we don't want to strictly lock this in here; otherwise no self-hosters would be able to work with any private Weblate instances.

We only set the value explicitly for things like SaaS-only services